lib/browser/guest-view-manager.js

```diff
@@ -303,9 +282,19 @@ const watchEmbedder = function (embedder) {
   }
   watchedEmbedders.add(embedder)
 
+  // The embedder may be manually destroyed earlier than guest.
+  embedder.once('will-destroy', () => {
+    for (const guestInstanceId in guestInstances) {
+      const guestInstance = guestInstances[guestInstanceId]
+      if (guestInstance.embedder === embedder) {
+        detachGuest(embedder, parseInt(guestInstanceId))
```

Comment:
Shouldn't the `embedder` be removed from `watchedEmbedders` and also the event listener `-window-visibility-change` for embedder ?